src/spikeinterface/sortingcomponents/waveforms/waveform_thresholder.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+import json
+from typing import List, Optional
+import scipy.signal
+import numpy as np
+
+from spikeinterface.core import BaseRecording
+from spikeinterface.sortingcomponents.peak_pipeline import (
+    PipelineNode,
+    WaveformExtractorNode,
+)
+
+
+class WaveformThresholder(WaveformExtractorNode):
+    """
+    Waveform Thresholder based on selected feature
+    This allow to perform an adaptive masking, by setting to 0 channels
+    that have a given feature below a given threshold
+
+    Parameters
+    ----------
+    feature: string in ['ptp', 'mean', 'energy', 'v_peak']
+        the feature that should be considered
+    threshold: float
+        the threshold below which channels are set to 0
+    """
+
```

Comment:
Thanks for adding the typing. My suggestion for improving the docstring:


    """
    A node that" performs waveform thresholding based on a selected feature.

    This node allows you to perform adaptive masking by setting channels to 0
    that have a given feature below a certain threshold. The available features
    to consider are peak-to-peak amplitude ('ptp'), mean amplitude ('mean'),
    energy ('energy'), and voltage peak ('v_peak').

    Parameters
    ----------
    recording: BaseRecording
        The recording extractor object.
    return_output: bool, optional
        Whether to return output from this node (default True).
    parents: list of PipelineNodes, optional
        The parent nodes of this node (default None).
    feature: {'ptp', 'mean', 'energy', 'v_peak'}, optional
        The feature to be considered for thresholding (default 'ptp').
    threshold: float, optional
        The threshold value for the selected feature (default 2).
    """
    
Am I understanding and describing the v-peak method appropiately?